### PR TITLE
[Feature] QR id param for link and analytics

### PR DESCRIPTION
**Pull request by Tysyatsky**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @MementoMorj 
- [ ] @NadiaKishchuk 
- [ ] @Kasterov 
- [ ] @Kotusyk 
- [ ] @Katerix 
- [ ] @MaksBrat 
- [ ] @Dobriyr 
- [ ] @DanyilTerentiev 
- [ ] @ormykhalyshyn 

### Second Level Review

- [ ] @LanchevychMaxym 

## Summary of issue

Creating front-end part for statistics for qr scans.

## Summary of change

Added param in url and sending requests to api.

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/features/AdminPage/NewStreetcode/MapBlock/StatisticsStreetcodeAdmin/StatisticsAdmin.styles.scss [hunk 1]
```diff
@@ -12,13 +12,21 @@
...
   box-shadow: 0px 7px 11px 2px rgba(0, 0, 0, 0.25);
   border-radius: 23px;
+  
+  .input-id{
+    @include mut.sized(300px, 40px);
+    @include mut.full-rounded(10px);
+    @include mut.rem-margined(16px, 0, 15px, 20px);
+    justify-content: space-between;
+    display: flex;
+    flex-direction: column;
```
Comment (MaksBrat) [inline src/features/AdminPage/NewStreetcode/MapBlock/StatisticsStreetcodeAdmin/StatisticsAdmin.styles.scss:22]: use mut.flexed

**Review by MaksBrat — APPROVED** [review]

File: src/app/router/Routes.tsx [hunk 2]
```diff
@@ -24,12 +25,12 @@ const router = createBrowserRouter(createRoutesFromElements(
         <Route
             index
             path={`${FRONTEND_ROUTES.ADMIN.BASE}`}
-            element={<ProtectedComponent><AdminPage /></ProtectedComponent>}
+            element={<AdminPage />}
```
Comment (Katerix) [inline src/app/router/Routes.tsx:28]: I am not sure if this is ment to be commited. Tell me if I am wrong here

File: src/app/router/Routes.tsx [hunk 3]
```diff
@@ -24,12 +25,12 @@ const router = createBrowserRouter(createRoutesFromElements(
...
         />
         <Route
             index
             path={FRONTEND_ROUTES.ADMIN.NEW_STREETCODE}
-            element={<ProtectedComponent><NewStreetcode /></ProtectedComponent>}
+            element={<NewStreetcode />}
         />
```
Comment (Katerix) [inline src/app/router/Routes.tsx:34]: same

File: src/features/AdminPage/NewStreetcode/MapBlock/MapAdmin/MapAdmin.component.tsx [hunk 4]
```diff
@@ -152,26 +153,51 @@ const MapOSMAdmin = () => {
...
             longtitude: item.longtitude,
-            city: item.city || cityName,
             actions: item,
         }),
     );
-    const handleLoadScriptNext = () => {
-        if (geocoderRef.current === null) {
-            geocoderRef.current = new google.maps.Geocoder();
+
+
```
Comment (Katerix) [inline src/features/AdminPage/NewStreetcode/MapBlock/MapAdmin/MapAdmin.component.tsx:166]: extra line

File: src/features/AdminPage/NewStreetcode/MapBlock/MapAdmin/MapAdmin.component.tsx [hunk 5]
```diff
@@ -152,26 +153,51 @@ const MapOSMAdmin = () => {
...
+                        message.error({
+                            content: 'Даний айді вже використовується. Використайте інший, будь ласка.',
+                            style: { marginTop: '400vh' }, 
+                          });
+                    } else {
+                        message.success({
+                            content:   'Такій айді вільний. Можете з впевненістю його використовувати',
+                            style: { marginTop: '400vh' }, 
+                          });
+
```
Comment (Katerix) [inline src/features/AdminPage/NewStreetcode/MapBlock/MapAdmin/MapAdmin.component.tsx:182]: same

File: src/features/AdminPage/NewStreetcode/MapBlock/MapAdmin/MapAdmin.component.tsx [hunk 6]
```diff
@@ -152,26 +153,51 @@ const MapOSMAdmin = () => {
...
+            message.error('Поле порожнє');
         }
     };
+
+    const handleNewNumberChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setNewNumber(event.target.value);
+      };
     return (
 
         <LoadScript
```
Comment (Katerix) [inline src/features/AdminPage/NewStreetcode/MapBlock/MapAdmin/MapAdmin.component.tsx:198]: same

File: src/features/AdminPage/NewStreetcode/MapBlock/MapAdmin/MapAdmin.component.tsx [hunk 7]
```diff
@@ -184,25 +210,40 @@ const MapOSMAdmin = () => {
             >
                 <div className="statisticsContainerAdmin">
                     <h1>Додати стріткод на мапу:</h1>
-                    <h1>{cityName}</h1>
+                    <h1>{address}</h1>
+                    
+  
+
                     <Autocomplete onLoad={onLoad} onPlaceChanged={onPlaceChanged}>
```
Comment (Katerix) [inline src/features/AdminPage/NewStreetcode/MapBlock/MapAdmin/MapAdmin.component.tsx:217]: same

File: src/features/AdminPage/NewStreetcode/MapBlock/MapAdmin/MapAdmin.component.tsx [hunk 8]
```diff
@@ -184,25 +210,40 @@ const MapOSMAdmin = () => {
...
+                        <Input
+                            type="number"
+                            className="input-streets"
+                            placeholder="введіть номер"
+                            onChange={handleNewNumberChange}
+                            value={newNumber}
+                        />
+                    
+
+                    <Button className="onMapbtn" onClick={onCheckIndexClick}>
```
Comment (Katerix) [inline src/features/AdminPage/NewStreetcode/MapBlock/MapAdmin/MapAdmin.component.tsx:234]: same

File: src/features/AdminPage/NewStreetcode/MapBlock/MapAdmin/MapAdmin.component.tsx [hunk 9]
```diff
@@ -184,25 +210,40 @@ const MapOSMAdmin = () => {
...
+
                     <Autocomplete onLoad={onLoad} onPlaceChanged={onPlaceChanged}>
                         <Input
                             className="input-streets"
                             placeholder="введіть вулицю"
                             prefix={<EnvironmentOutlined className="site-form-item-icon" />}
                         />
                     </Autocomplete>
+
+                        <Input
```
Comment (Katerix) [inline src/features/AdminPage/NewStreetcode/MapBlock/MapAdmin/MapAdmin.component.tsx:225]: same

**Review by Katerix — COMMENTED** [review]
good job! I approve this pr, but please before merging it check the comments below to avoid unpleasant mistakes

**Review by Katerix — APPROVED** [review]

**Review by Katerix — APPROVED** [review]
lgtm

**Pull request closed without merging**